Spanish Translations for remaining untranslated UI components (#12427)
galaxyproject/galaxy: doc/source/dev/translating.rst

Translating the Interface
=========================

A quick tutorial on translating the Galaxy UI
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Translations in Galaxy are done by simple dictionary lookup of text. Whatever text is written in the interface, the client will look it up in its dictionary, and if it's there, return it. If it's not, it will return the english text that was used to search.

Reviewer: ```suggestion
Translations in Galaxy are done by simple dictionary lookup of text. Whatever text is written in the interface, the client will look it up in its dictionary, and if it's there, return it. If it's not, it will return the English text that was used to search.
```

> Translations are stored in ``client/src/nls/*/locale.js`` and look like this:

Reviewer: ```suggestion
Translations are stored in ``client/src/nls/XX/locale.js`` files where `XX` is the ISO 639 2-letter code for a language. A `locale.js` file looks like this:
```

Not sure that's correct, but I believe that's how it works?

Author: if not close enough, the audience of this document is non-technical.

.. code-block:: js

    define({
        // ----------------------------------------------------------------------------- masthead

        "Analyze Data": "Analizar Datos",
        "Tools and Current History": "Herramientas e Historial Actual",
        Workflow: "Flujo de Trabajo",

The last three lines contain a mapping of English text to their Spanish equivalents. But before we can add translations here, we need to know what English text is being used in the interface. Generally this is a process of:

- Find untranslated text in the UI
- Find the corresponding text in the codebase

> - Ensure that it uses `_l` (localisation) statements

Reviewer: ```suggestion
- Ensure that it uses `_l()` localisation statements
```

> - Check that that text has a translation in the locale file.

Reviewer: ```suggestion
- Add the translation for that text to the `locale.js` file.
```

A quick example.
~~~~~~~~~~~~~~~~

In the user preferences menu I saw the text: "You are logged in as hxr@local.host." I can guess that the email address is templated out, so I'm going to search for "You are logged in as". Running that query I see some results:

.. code-block:: shell

>     $ grep 'You are logged in as' -R client/src

Reviewer: ```suggestion
    $ grep -R 'You are logged in as' client/src/
```

    client/src/mvc/library/library-library-view.js:                        You are logged in as an <strong>administrator</strong> therefore you can manage any library
    client/src/components/Libraries/LibraryFolder/LibraryFolderPermissions/LibraryPermissionsWarning.vue:                You are logged in as an <strong>administrator</strong> therefore you can manage any folder on this
    client/src/components/User/UserPreferences.vue:            You are logged in as <strong id="user-preferences-current-email">{{ email }}</strong

There! That's a hit, the last file has the string we're looking for. Let's open that file and discover the next challenge...

Types of Strings and how to translate them
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> There are a number of ways text gets used in the codebase, especially the new Vue frontend.

Reviewer: ```suggestion
There are a number of ways text gets used in the codebase, especially the new Vue JavaScript framework.
```
